### DISCUSSION TITLE
docking station?

### DISCUSSION THREAD

**Discussion by adalbehrt** [SkyrilHD/Dell-E7250-Hackintosh#59] [General]
Hi!

Thanks for your wonderful install package! 
I was wondering... is it possible to get the docking port/docking station working? It's about the only thing I've been missing!

**Comment by SkyrilHD** [comment]
Docking station works for me OOTB. However, I did not test display outputs as I don't have a DisplayPort monitor. Audio and USB are working fine.

**Reply by adalbehrt** [reply]
Well, this is the issue. E7250 is a nice machine, but it has the old style docking port, rather than USB-C. No signal seems to be coming out of the display port on the docking station once the laptop is docked. HDMI on the laptop works, but only after the machine is undocked. Not a big problem -- one can just skip the docking station altogether, but I was just wondering whether there was any possibility to reroute hdmi-signal through the display port.

**Reply by SkyrilHD** [reply]
I will get an adapter and try it.

**Reply by adalbehrt** [reply]
Please do if you have a chance, and tell us how it goes.

**Reply by SkyrilHD** [reply]
So, after some testing, I can confirm that display output via docking station does not work. However, I get a reaction from macOS (which is good, I guess?) that it tries to initialize a new display (screen goes black for a second), but it fails. Checking IORegistryExplorer, I see a new 'AppleDisplay' entry, which means that 'a' screen is recognized. Checking System-Profiler and System Preferences, it (obviously) does not show anything. My monitor only accepts DVI, so I got a DP -> DVI adapter. And connecting via adapter or directly to the dock did not show any different results, but still fails both ways to initialize the display.

To confirm, it is not due to my laptop or my dock, I tried the same steps on Windows. To no one's surprise, it works.

So, my next troubleshooting step was booting to macOS with the lid closed. Verbose mode over the external display works fine until the last boot step (basically the moment before verbose mode gets dismissed and the Apple logo shows up). It refuses to leave verbose mode and gets stuck there. Probably because, again, it cannot initialize the display. Weirdly enough, macOS is still running fine in the background, which means macOS does not kernel panic or hang.

I tried changing some entries inside config.plist, like the framebuffer connector type or disabling AGDP, etc. At this point, I need to do some research to fix this issue. If you have any founding, it would be great to gather it here.
